### PR TITLE
Indirect QuickRun - add diffusion option

### LINKED ISSUE #18223: Indirect QuickRun - Diffraction
Add an option to run through IndirectDiffractionReduction to existing EnergyWindowScan algorithm. The diffraction should only be in diffspec mode and it is assumed that default inputs will be used, so the users shouldn't have to modify them.

### PR DESCRIPTION
**Pull request by louisemccann**
EnergyWindowScan has been extended to include diffraction reduction and renamed to IndirectReductionAndAnalysis to better represent what the algorithm does.

**To test:**
InputFiles: [Diff_inputs.zip](https://github.com/mantidproject/mantid/files/774174/Diff_inputs.zip)

1
Interfaces>Indirect>Diffraction
Input: `26181` 
(this should find the appropriate run in the data archive otherwise browse to `IRS26181` in files provided)
Ensure Instrument = `ISIS` and mode = `diffspec`
Press `Run`

2
```Python
IndirectReductionAndAnalysis(InputFiles='26176',
                             Instrument='IRIS',
                             Analyser='graphite',Reflection='002',
                             SpectraRange='3, 50',
                             ElasticRange='-0.5, 0',
                             InelasticRange='0, 0.5',
                             GroupingMethod='All',
                             MSDFit=False,
                             Diffraction=True,
                             DiffractionSpectraRange=[105,112])
```
Ensure irs26176_diffspec_red is the same for both methods

Fixes #18223 #12270

[Release Notes](https://github.com/mantidproject/mantid/blob/be1b1486734488150de3569407f561772a855007/docs/source/release/v3.10.0/indirect_inelastic.rst)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by samueljackson92** [comment]
@louisemccann Question: `ElasticWindowScan` is already an algorithm in 3.9 right? So by renaming the algorithm to `IndirectReductionAndAnalysis` does this break the Python API? `ElasticWindowScan` is pretty niche, but shouldn't we deprecate for a version then remove?

One option would be to just point the deprecated `ElasticWindowScan` at the new algorithm.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectReductionAndAnalysis.py [hunk 1]
```diff
@@ -213,8 +247,22 @@ def PyExec(self):
...
+            diff_red_alg.setProperty('ContainerFiles',self._can_files)
+            diff_red_alg.setProperty('SumFiles', self._sum_files)
+            diff_red_alg.setProperty('LoadLogFiles', self._load_logs)
+            diff_red_alg.setProperty('Instrument', self._instrument_name)
+            diff_red_alg.setProperty('Mode', self._diff_mode)
+            diff_red_alg.setProperty('SpectraRange',self._diff_spec)
+            diff_red_alg.setProperty('GroupingPolicy', self._grouping_method)
+            diff_red_alg.setProperty('OutputWorkspace', self._diff_workspace)
+            diff_red_alg.execute()
+            self.setProperty("DiffractionWorkspace", diff_red_alg.getProperty("OutputWorkspace").value)
```
Comment (samueljackson92) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectReductionAndAnalysis.py]: All of the stuff under the if statement could be refactored into a method.

Comment (samueljackson92) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectReductionAndAnalysis.py]: In fact I'd go for one method per algorithm i.e. something like `self.run_diffraction_reduction()`, `self.run_msg()` ...

**Review by samueljackson92 — CHANGES_REQUESTED** [review]
See suggested changes above, or come talk to me if you disagree.

**Comment by SpencerHowells** [comment]
We need EnergyWindowScan for use by the Interface in #18216

**Comment by samueljackson92** [comment]
@SpencerHowells I agree the algorithm needs to exist. My problem is with renaming. Renaming would potentially break any user scripts that we have no control over that use the current name. Hence we should really deprecate the name `EnergyWindowScan` name for one version of Mantid, then remove any reference of the algorithm called `EnergyWindowScan` in favour of `IndirectReductionAndAnalysis`.

**Comment by louisemccann** [comment]
It has been decided that EnergyWindowScan and the DiffractionScan should be 2 different algs so I am going to close this.

**Pull request closed without merging**